File: digito-verificador-ci.php
#!/usr/bin/php
<?php
function calculo_digito($cedula){
	 $num="2987634";
	 $acumulo = 0;
	 $error = "";
         $sinpunto=str_replace(".","",$cedula);
         $singuion=str_replace("-","",$sinpunto);
         $sindigito=substr($singuion,0,7);
         $sindigito=str_replace(" ","",$sindigito);
         if(strlen($sindigito)<6){
         	return "el numero de cédula no tiene un formato válido";
       	 }
         $digito=strrev($cedula);
         $digito=substr($digito,0,1);
         for($x=0;$x<7;$x++){
         		$n=substr($sindigito,$x,1);
         		$y=substr($num,$x,1);
                $resultado=$n*$y;
                $resultado=10-substr(strrev($resultado),0,1);
                $acumulo=$acumulo+$resultado;
         }
         $xdigito=substr(strrev($acumulo),0,1);
         return $xdigito;
}
if (isset($argv[1])) {
	$cedula = $argv[1];
	$digito_verificador=calculo_digito($cedula);
	if (is_numeric($digito_verificador) ) {
	     print $cedula.$digito_verificador;
	} else if ((!isset($argv[2]))) {
	     print $digito_verificador;
	}
	if (!isset($argv[2])) {
		print "\n";
	}
} else {
	//Imprimo ayuda
	print "Uso:\n";
	print "    digito-verificador-ci cedula [q]\n";
	print "q = Modo silencioso (para ser usado desde un script)\n";
	print "ejemplo: \n";
	print "         digito-verificador-ci 1234567\n";
}
?>

File: digito-verificador-ci.py
import sys

def calculo_digito(cedula):
	num="2987634"
	acumulo = 0
	error = ""
	sinpunto=cedula.replace(".","")
	singuion=sinpunto.replace("-","")
	sindigito=singuion[0:7]
	sindigito=sindigito.replace(" ","")
	if(len(sindigito)<6):
		return "el numero de cedula no tiene un formato valido"       	 
	digito=cedula[::-1]
	digito=digito[0]
	for x in range(0,7):
 		n=sindigito[x]
         	y=num[x]
		resultado=int(n)*int(y)
		resultado=10-int(str(resultado)[::-1][0])
		acumulo=acumulo+resultado      
	xdigito=str(acumulo)[::-1][0]
	return xdigito
try :
	cedula = sys.argv[1]
	digito_verificador=calculo_digito(cedula)
	if (unicode(digito_verificador, 'utf-8').isnumeric()) :
		print cedula+digito_verificador
	else:
		print digito_verificador	
except NameError :
	print "  Uso:\n"
	print "         digito-verificador-ci numcedula\n\n"
	print "  ejemplo: \n"
	print "         digito-verificador-ci 1234567\n"

